jvm/src/main/java/com/microsoft/bond/protocol/TaggedProtocolReader.java

```diff
@@ -0,0 +1,75 @@
+package com.microsoft.bond.protocol;
+
+import com.microsoft.bond.BondDataType;
+
+import java.math.BigInteger;
+
+/**
+ * Reads a serialized payload.
+ */
+public interface TaggedProtocolReader {
+    final class ReadFieldResult {
```

Comment:
Since you apparently don't need the enclosing instance, please make this both inner classes static.